astropy/time/core.py

```diff
@@ -2285,7 +2285,13 @@ def _make_array(val, copy=False):
     # Allow only float64, string or object arrays as input
     # (object is for datetime, maybe add more specific test later?)
     # This also ensures the right byteorder for float64 (closes #2942).
-    if not (val.dtype == np.float64 or val.dtype.kind in 'OSUMaV'):
+    if val.dtype == np.float64:
+        pass
+    elif val.dtype.kind in 'OSUMaV':
+        pass
+    elif val.dtype.kind == "f" and val.dtype.itemsize > np.dtype(np.float64).itemsize:
```

Comment:
My sense would be just not to force at all for this case. If someone implemented a custom high-precision float dtype, then good for them!

Reply:
Yeah, I'm not sure I understand why this function is doing any type coercion at all so I'm hesitant to remove it. Well okay the existing type checker code chokes on integers, so we need to at least up-convert those. And I'm worried about the endianness thing, but the answer there is more tests!

Reply:
That coercion was definitely there for integers and `float32` (and failure on complex, I think).

While testing may not be 100% complete, it is meant to be good enough that one can experiment and trust that things are not too bad if all tests pass...

Reply:
Bleh. Lots of existing code chokes if I remove the byte swapping for doubles, and it's not so easy to see whether the underlying problem (ERFA might fail with wrong-endian data?) remains. So I still normalize the byte order of float64s.

Reply:
Hmm, if ERFA fails then I didn't set the ufuncs up right. Could you make a separate issue for that?

Reply:
Looking at #2942 I think the problem might just have been that astopy.time.core required a particular endianness from floats; no actual ERFA error reports are shown. I'll try relaxing the input validation and see what it breaks. I suspect nothing.

Reply:
Indeed all tests still pass once I remove the endianness check during construction.